WooCommerce/Classes/ViewRelated/Products/Variations/Bulk Update/BulkUpdateViewModel.swift

```diff
@@ -93,3 +177,39 @@ extension BulkUpdateViewModel {
         static let numberOfObjects = 100
     }
 }
+
+private extension BulkUpdateViewModel {
+    /// Represents if a property in a collection of `ProductVariation`  has the same value or different values or is missing.
+    ///
+    enum BulkValue {
+        /// All variations have the same value
+        case value(String)
+        /// When variations have mixed values.
+        case mixed
+        /// None of the variation has a value
+        case none
+    }
+
+    /// Calculates and returns if a property, specified by a `ProductVariation` keypath , has the same value or different values or is missing
+    /// over all product variations of this view model.
+    ///
+    func bulkValueOf(_ keypath: KeyPath<ProductVariation, String?>) -> BulkValue {
```

Comment:
I used a keypath since in the initial implementation I reused it for the sale price also. Makes it more reusable for other bulk update options also.